master/reference/calicoctl/setup/etcdv2.md

```diff
@@ -1,132 +1,63 @@
-
-```
-$ etcdctl role add calico-readwrite
-$ etcdctl role grant calico-readwrite -path '/calico' -readwrite
-$ etcdctl role grant calico-readwrite -path '/calico/*' -readwrite
+etcdEndpoints: http://etcd1:2379,http://etcd2:2379 # A comma separated list of etcd endpoints.
+etcdUsername: user                 # Username for RBAC (optional)
+etcdPassword: password             # Password for the given username (optional)
+etcdKeyFile: /etc/calico/key.pem   # Path to the etcd client key file (optional)
+etcdCertFile: /etc/calico/cert.pem # Path to the etcd client certificate file (optional)
+etcdCACertFile: /etc/calico/ca.pem # Path to the etcd CA certificate file (optional)
 ```
 
-#### Revoke write access for the guest user
-
-Revoke write access for the guest user.  This means the non-authenticated user of
-the etcd cluster will have read-only access.
-
-```
-$ etcdctl role revoke guest -path '/*' -write
-```
-
-### Configure users
+> **NOTE:**
+> If you are running with TLS enabled, ensure your endpoint addresses use https
 
-#### Configure calicooctl user
-Create a calicoctl user and enter your chosen password.
+## Environment Variables
 
-```
-$ etcdctl user add calicoctl
-New password:
-```
+Each of the above configuration options can also be set through environment variables.  For example,
+`etcdEndpoints` can also be represented as `ETCD_ENDPOINTS`.
 
-#### Assign the calicoctl role to this user
 
-```
-$ etcdctl user grant calicoctl -roles calico-readwrite
-```
+#### Root Environment Variables
 
-#### Create the root user
-Before enabling authentication, it is also necessary to create a root user for the
-cluster.  
+Some commands will require root access to run. By default, the root user's environment
+will not inherit the environment variables set in the original user's environment.
+ The user's environment variables
+specified above will not be recognized by the root user, so the variables must
+be passed into the Calico command.
 
-Create the root user and enter your chosen password
+If you have previously defined/exported your environment
+variables, you can run `sudo` with the `-E` flag to pass in your environment:
 
-```
-$ etcdctl user add root 
-New password:
+```shell
+sudo -E calicoctl node run
 ```
 
-### Enable authentication
+Another common solution is directly pass environment variables to the commands
+run as root. The following example shows how to pass configuration settings
+to the `node run` command:
 
-Finally enable authentication
-
-```
-$ etcdctl auth enable
+```shell
+sudo ETCD_KEY_FILE=/path/to/client.key \
+     ETCD_CA_CERT_FILE=/path/to/ca.crt ETCD_CERT_FILE=/path/to/client.crt \
+     ETCD_ENDPOINTS=http://hostname:2379 calicoctl node run
 ```
 
-Your etcd is now running with authentication enabled. To disable it for any reason, 
-use the reciprocal command:
-
-```
-$ etcdctl -u root:<rootpw> auth disable
-```
+## calico/node
```

Comment:
I think this belongs in the `calico/node` doc (which may not exist yet).

This document is about configuring calicoctl to interact with an etcdv2 cluster.